### DISCUSSION TITLE
ADS-DV CAN Communication | Error in ioctl(): No such device

### DISCUSSION THREAD

**Discussion by nihadjifrikonathputhiyamaliyakkal** [FS-AI/FS-AI_2020#6] [General]
Hi Team,

While running the ros_can node on ADS-DV, we continue to have the ioctl problem. (No such Device) even after `can-utils` is installed. could you kindly guide us more on this?

![image](https://github.com/FS-AI/FS-AI_Compute/assets/19814532/0bd63fb9-bde9-4fb8-bc2d-b61d1ac05451)

Greatly appreciate it!

**Comment by IanMurphy-StreetDrone** [comment]
You have to run a command similar to 'sudo ip link set up can0 type can bitrate 500000' every time the OS boots, to bring up a valid CAN device at the correct bit rate. This has nothing to do with 'can-utils'. You also need to add the kernel modules on install.

See https://github.com/FS-AI/FS-AI_API/blob/main/setup.sh
